packages/table/src/regions.ts

```diff
@@ -548,11 +541,52 @@ export class Regions {
         }
     }
 
+    private static overlapsRegionInternal(regions: IRegion[], query: IRegion, allowPartialOverlap = true) {
+        const intervalCompareFn = allowPartialOverlap ? Regions.overlapsRegion : Regions.containsRegion;
+
+        if (regions == null || query == null) {
+            return false;
+        }
+
+        for (const region of regions) {
+            const cardinality = Regions.getRegionCardinality(region);
+            switch (cardinality) {
+                case RegionCardinality.FULL_TABLE:
+                    return true;
+                case RegionCardinality.FULL_COLUMNS:
+                    if (intervalCompareFn(region.cols, query.cols)) {
+                        return true;
+                    }
+                    continue;
+                case RegionCardinality.FULL_ROWS:
+                    if (intervalCompareFn(region.rows, query.rows)) {
+                        return true;
+                    }
+                    continue;
+                case RegionCardinality.CELLS:
+                    if (intervalCompareFn(region.cols, query.cols)
+                        && intervalCompareFn(region.rows, query.rows)) {
+                        return true;
+                    }
+                    continue;
+                default:
+                    break;
+            }
+        }
+
+        return false;
+    }
+
     private static regionsEqual(regionA: IRegion, regionB: IRegion) {
         return Regions.intervalsEqual(regionA.rows, regionB.rows)
             && Regions.intervalsEqual(regionA.cols, regionB.cols);
     }
 
+    private static regionContains(regionA: IRegion, regionB: IRegion) {
+        // containsRegion expects an array of regions as the first param
+        return Regions.overlapsRegionInternal([regionA], regionB, /* allowPartialOverlap */ false);
```

Comment:
_At least_ partial in my opinion, which means something could fully overlap another. If you can think of better terminology, do feel free to suggest it! I'm not in love with these names.